pkg/test/setup/setup.go

```diff
@@ -236,48 +233,58 @@ func startDataNode(etcdEndpoint string, flags ...string) (string, string, func()
 		return helpers.ListKeys(etcdEndpoint, fmt.Sprintf("/%s/nodes/%s:%d", metadata.DefaultNamespace, nodeHost, ports[0]))
 	}, testflags.EventuallyTimeout).Should(gomega.HaveLen(1))
 
-	return addr, path, closeFn, deferFn
+	return addr, dataDir, closeFn
 }
 
 // DataNode runs a data node.
 func DataNode(etcdEndpoint string, flags ...string) func() {
-	_, _, closeFn, deferFn := startDataNode(etcdEndpoint, flags...)
+	path, deferFn, err := test.NewSpace()
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	closeFn := DataNodeFromDataDir(etcdEndpoint, path, flags...)
 	return func() {
 		closeFn()
 		deferFn()
 	}
 }
 
+// DataNodeFromDataDir runs a data node with a specific data directory.
+func DataNodeFromDataDir(etcdEndpoint, dataDir string, flags ...string) func() {
+	_, _, closeFn := startDataNode(etcdEndpoint, dataDir, flags...)
+	return closeFn
+}
+
 // DataNodeWithAddrAndDir runs a data node and returns the address and root path.
 func DataNodeWithAddrAndDir(etcdEndpoint string, flags ...string) (string, string, func()) {
-	addr, dir, closeFn, deferFn := startDataNode(etcdEndpoint, flags...)
+	path, deferFn, err := test.NewSpace()
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	addr, dir, closeFn := startDataNode(etcdEndpoint, path, flags...)
 	return addr, dir, func() {
 		closeFn()
 		deferFn()
 	}
 }
 
 // LiaisonNode runs a liaison node.
-func LiaisonNode(etcdEndpoint string, flags ...string) (string, func()) {
+func LiaisonNode(etcdEndpoint string, flags ...string) (grpcAddr, httpAddr string, closeFn func()) {
```

Comment:
Could you add a "LiaisonNodeWithHTTP" to expose the httpAddr instead of changing this function's signature?